src/net/tcp/stream.rs

```diff
@@ -87,10 +87,10 @@ impl TcpStream {
     /// entries in the routing cache.
     ///
     /// [write interest]: Interest::WRITABLE
-    #[cfg(not(target_os = "wasi"))]
+    #[cfg(any(not(target_os = "wasi"), target_env = "p2"))]
```

Comment:
Can you use `not(all(...))` here?

Reply:
If you mean `not(all(target_os = "wasi", target_env = "p1"))` then the same rationale I gave above for not using `target_env = "p1"` applies.  Again, though. happy to change it if we're not concerned about backwards `rustc` compatibility.